toml/src/main/kotlin/org/rust/toml/crates/local/CratesLocalIndexService.kt

```diff
@@ -181,22 +214,32 @@ class CratesLocalIndexService : PersistentStateComponent<CratesLocalIndexState>,
         }
     }
 
-    private inner class CratesLocalIndexUpdateTask : Task.Backgroundable(null, "Loading cargo registry index", false) {
+    private inner class CratesLocalIndexUpdateTask(val newHead: String) : Task.Backgroundable(null, "Loading cargo registry index", false) {
         override fun run(indicator: ProgressIndicator) {
-            val branch = repository.resolve("origin/master") ?: return
-            val revCommit = RevWalk(repository).parseCommit(ObjectId.fromString(branch.name))
-            reloadCrates(revCommit.tree)
+            updateCrates(newHead, state.indexedCommitHash)
         }
 
         override fun onSuccess() {
-            state = CratesLocalIndexState(registryHeadCommitHash)
+            state.indexedCommitHash = newHead
         }
 
         override fun onFinished() {
             isReady.set(true)
         }
     }
 
+    @Suppress("unused")
+    private class CargoTomlEventsListener : BulkFileListener {
+        override fun after(events: MutableList<out VFileEvent>) {
+            if (!events.any { it.path.endsWith(CargoConstants.MANIFEST_FILE) }) return
+
+            val cratesService = service<CratesLocalIndexService>()
```

Comment:
I think this listener should be attached using [`MessageBus`](https://jetbrains.org/intellij/sdk/docs/reference_guide/messaging_infrastructure.html), probably like that
```Kotlin
init {
    ApplicationManager.getApplication().messageBus.connect().subscribe(VirtualFileManager.VFS_CHANGES, object : BulkFileListener {
        override fun after(events: MutableList<out VFileEvent>) { ... }
    })
}
```

Please check `org.rust.cargo.project.model.impl.CargoProjectsServiceImpl` for another example.